stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -136,6 +136,8 @@ internal class StripePaymentController internal constructor(
         confirmStripeIntentParams: ConfirmStripeIntentParams,
         requestOptions: ApiRequest.Options
     ) {
+        logReturnUrl(confirmStripeIntentParams.returnUrl)
+
         val returnUrl = confirmStripeIntentParams.returnUrl.takeUnless { it.isNullOrBlank() }
             ?: defaultReturnUrl.value
```

Comment:
can we combine the `when` clause inside as `#logReturnUrl` is not reused elsewhere?

Reply:
Hm not sure I understand your request

Reply:
oh I meant inline `logReturnUrl ` here and use a single `when` clause to send analyticsRequest and assign value for `reutrnUrl`

Reply:
I'm still not following, can you show me a code example? 🙃

Reply:
Sure, something like
```
val (returnUrl, event) = when (confirmStripeIntentParams.returnUrl) {
    defaultReturnUrl.value -> {
        Pair(confirmStripeIntentParams.returnUrl, AnalyticsEvent.ConfirmReturnUrlDefault)
    }
    null -> {
        Pair(defaultReturnUrl.value, AnalyticsEvent.ConfirmReturnUrlNull)
    }
    else -> {
        Pair(confirmStripeIntentParams.returnUrl, AnalyticsEvent.ConfirmReturnUrlCustom)
    }
}

analyticsRequestExecutor.executeAsync(
    analyticsRequestFactory.createRequest(event)
)
```

Reply:
I'd rather keep the logging logic separate - it's not directly related to the main function of the method.